solcore/optimization/differential_evolution.py

```diff
@@ -0,0 +1,158 @@
+"""
```

Comment:
I've just realized that you have copied from the original repo the whole classes that you want to replace, and then tweak those parts that you don't like. I think that a better approach will be to create subclasses of the original classes and then override the relevant methods/attributes you want. Methods and atributes not overriden will be inherited from the original implementation.

Something like:
```python
class MyClass(ClassToReplace):
    def method_to_replace(self, p1, p2):
        "your implementation of the method"
        return whatever_you_like
```

Here `MyClass` will be identical to `ClassToReplace` except for the `method_to_replace`.

This way, it becomes clear what are the differences with the original code, makes your own code way shorter and, in summary, easier to debug.

Reply:
good idea, thank you!